app/src/main/java/chat/rocket/android/createChannel/ui/CreateNewChannelActivity.kt

```diff
@@ -0,0 +1,187 @@
+import android.os.Bundle
+import android.support.design.chip.Chip
+import android.support.v7.app.AppCompatActivity
+import android.view.MenuItem
+import chat.rocket.android.R
+import chat.rocket.android.createChannel.addMembers.ui.AddMembersActivity
+import chat.rocket.android.createChannel.presentation.CreateNewChannelPresenter
+import chat.rocket.android.createChannel.presentation.CreateNewChannelView
+import chat.rocket.android.util.extensions.setVisible
+import chat.rocket.android.util.extensions.showToast
+import chat.rocket.common.model.roomTypeOf
+import com.jakewharton.rxbinding2.widget.RxTextView
+import dagger.android.AndroidInjection
+import kotlinx.android.synthetic.main.activity_create_new_channel.*
+import kotlinx.android.synthetic.main.layout_toolbar.*
+import javax.inject.Inject
+
+internal const val ADD_MEMBERS_ACTIVITY_REQUEST_CODE = 1
+
+class CreateNewChannelActivity : AppCompatActivity(), CreateNewChannelView {
+    @Inject
+    lateinit var presenter: CreateNewChannelPresenter
+    private var channelType: String = "public"
+    private var listOfUsers: ArrayList<String> = ArrayList()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        AndroidInjection.inject(this)
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_create_new_channel)
+        setUpToolBar()
+        setUpOnClickListeners()
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
+        when (item?.itemId) {
+            android.R.id.home -> {
+                finish()
+                return true
+            }
+
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (resultCode == Activity.RESULT_OK) {
+            if (requestCode == ADD_MEMBERS_ACTIVITY_REQUEST_CODE && data != null) {
+                listOfUsers = data.getStringArrayListExtra("members")
+                selected_members_chips.removeAllViews()
+                refreshMembersChips()
+            }
+        }
+    }
+
+    override fun showLoading() {
+        view_loading.setVisible(true)
+        layout_container.alpha = 0.5f
+        layout_container.isEnabled = false
+    }
+
+    override fun hideLoading() {
+        view_loading.setVisible(false)
+        layout_container.alpha = 1.0f
+        layout_container.isEnabled = true
+    }
+
+    override fun showChannelCreatedSuccessfullyMessage() {
+        showToast(getString(R.string.msg_channel_created_successfully))
+        finish()
+    }
+
+    override fun showMessage(resId: Int) {
+        channel_name_edit_text.setText("")
+        showToast(getString(resId))
+    }
+
+    override fun showMessage(message: String) {
+        channel_name_edit_text.setText("")
+        showToast(message)
+    }
+
+    override fun showGenericErrorMessage() {
```

Comment:
`ShowMessageAndClearTest()`? It is better to create another function and avoid doing things that a function isn't made for.